### DISCUSSION TITLE
New check: all necessary env variables are set

### DISCUSSION THREAD

**Discussion by eleftrik** [spatie/laravel-health#81] [Ideas]
I think checking if all needed **.env variables are set** could be useful (at least in my projects / deploy scripts 😅).
Before using spatie/laravel-health, I used antonioribeiro/health or beyondcode/laravel-self-diagnosis, and both implements this check.
The user should be able to specify which variable names should be checked, also taking into account the environment.

One simple example: when I'm on `production` environment, I need to check if `BUGSNAG_API_KEY` has been defined (i.e. it's not empty); this is (or could be) not needed in `local` environment.

Might it be worth implementing such a check?
Or should I rely on some other package dedicated to this task? (disclaimer: not meant to be a criticism).

**Comment by freekmurze** [comment]
Feel free to create a separate package where you create and maintain the check. When you're done, I'll be more than happy to add a link to it in our docs.

**Reply by eleftrik** [reply]
I'll give it a try. Thank you @freekmurze!

**Comment by eleftrik** [comment]
Hi @freekmurze, here we go: https://github.com/encodia/laravel-health-env-vars

Please, let me know if there's something I can improve/fix.

Thanks!

**Reply by freekmurze** [reply]
Nice, I've added this to our docs: https://github.com/spatie/laravel-health/blob/main/docs/available-checks/overview.md

**Reply by eleftrik** [reply]
@freekmurze Is it possible to add it to https://spatie.be/docs/laravel-health/v1/available-checks/overview also? Thanks!

**Reply by freekmurze** [reply]
Done!

**Reply by eleftrik** [reply]
Is it possible also to add a link in the sidebar, under AVAILABLE CHECKS, just like "SSL Certification expiration"? 
Thanks for your patience 😄

**Reply by freekmurze** [reply]
To decrease maintenance burden I've removed the other check from the sidebar. I think a link to third party repos is enough 👍

**Reply by eleftrik** [reply]
I agree with you. Thanks!
